crates/ra_ide_api/src/hover.rs

```diff
@@ -162,55 +163,55 @@ fn hover_text_from_name_kind(
 }
 
 pub(crate) fn hover(db: &RootDatabase, position: FilePosition) -> Option<RangeInfo<HoverResult>> {
-    let parse = db.parse(position.file_id);
-    let file = parse.tree();
+    let file = db.parse_or_expand(position.file_id.into())?;
+    let token = file.token_at_offset(position.offset).filter(|it| !it.kind().is_trivia()).next()?;
```

Comment:
We really need *some* set of helpers to avoid these `filter(|it| !it.kind().is_trivia())`, but I don't know what that set would be...